runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -42,120 +59,211 @@ public final class LogReplicationUtils {
+        };
 
-        try {
-            return IRetry.build(IntervalRetry.class, () -> {
-                try (TxnContext txnContext = corfuStore.txn(namespace)) {
-                    // The transaction is started in the client's namespace and the Replication Status table resides in the
-                    // system namespace.  Corfu Store does not validate the cross-namespace access as long as there are no
-                    // writes on the table in the different namespace.  This hack is required here as we want client full
-                    // sync to happen in the same transaction which checks the status of a snapshot sync so that there is no
-                    // window between the check and full sync.
-                    List<CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message>> entries =
-                        txnContext.executeQuery(replicationStatusTable,
-                                entry -> entry.getKey().getSubscriber().getModel()
-                                    .equals(LogReplication.ReplicationModel.LOGICAL_GROUPS) &&
-                                    Objects.equals(entry.getKey().getSubscriber().getClientName(),
-                                        clientListener.getClientName()));
-
-                    CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message> entry = null;
-
-                    // It is possible that there is no entry for the Logical Group Model in the status table in
-                    // the following scenarios:
-                    // 1. Request to subscribe is received before the Log Replication JVM or pod starts and
-                    // initializes the table
-                    // 2. Topology change which removes this cluster from the status table.
-                    // We cannot differentiate between the two cases here so continue with the right
-                    // behavior for the 1st case, i.e., subscribe and wait for the initial Snapshot Sync to get
-                    // triggered.  If we are here because of the second case, the listener will not get any updates
-                    // from LR and subscription will be a no-op.
-                    if (entries.isEmpty()) {
-                        log.info("No record for client {} and Logical Group Model found in the Status Table.  " +
-                                "Subscription could have been attempted before LR startup.  Subscribe the listener and" +
-                                "wait for initial Snapshot Sync", clientListener.getClientName());
-                    } else {
-                        // For a given replication model and client, any Sink node will have a single session.
-                        Preconditions.checkState(entries.size() == 1);
-                        entry = entries.get(0);
-                    }
+        private void readConfig() {
+            String pluginConfigFilePath = "src/test/resources/transport/pluginConfig.properties";
```

Comment:
This will work on Github corfu but not on a real setup.

The right way to get the config is https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java#L188

However, this code is running in the clients JVM/pod so we do not have access to the 'plugin' param.  Let us decide offline how it can be obtained from the client.  Either 1)pass the parameter in the client's startup file or 2)take it as an argument in the 'subscribe' call